core/src/main/scala/org/apache/spark/sql/SmartConnectorHelper.scala

```diff
@@ -74,6 +80,10 @@ class SmartConnectorHelper(snappySession: SnappySession) extends Logging {
     createUDFStmt = conn.prepareCall(createUDFString)
     dropUDFStmt = conn.prepareCall(dropUDFString)
     alterTableStmt = conn.prepareCall(alterTableStmtString)
+    getJarsStmt = conn.prepareCall(getJarsStmtString)
+    if (sc != null) {
+      executeGetJarsStmt(sc)
```

Comment:
IMO it should be in a try catch. Log a warning if something goes bad in terms of deployed jars. Failing it seems a littlel too restrictive.

Reply:
If you see presence of jar is not mandatory. For that it logs only a warning. But failure will indicate connection level problem which can affect smart connection otherwise so I don't think explicit handling is necessary.